File: rb/features.rb
# YamlYugi SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module YamlYugiFeatures
  def self.make_feature(name)
    case name
    when "base"
      YamlYugiBaseFeature.new
    when "test"
      YamlYugiTestFeature.new
    else
      YamlYugiBaseFeature.new
    end
  end
end

File: py/features.py
# YamlYugi SDK feature factory

from feature.base_feature import YamlYugiBaseFeature
from feature.test_feature import YamlYugiTestFeature


def _make_feature(name):
    features = {
        "base": lambda: YamlYugiBaseFeature(),
        "test": lambda: YamlYugiTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
